rust/src/guess.rs

```diff
@@ -105,64 +154,90 @@ impl Guess {
                 num_russian += 1;
             }
         }
-        let mut best_num = 0;
-        if num_dutch > 0 {
-            info!("Found {} dutch resources", num_dutch);
-            if best_num < num_dutch {
-                best_num = num_dutch;
-                best_version = Some(VanillaVersion::DUTCH);
-            }
-        }
-        if num_german > 0 {
-            info!("Found {} german resources", num_german);
-            if best_num < num_german {
-                best_num = num_german;
-                best_version = Some(VanillaVersion::GERMAN);
-            }
-        }
-        if num_italian > 0 {
-            info!("Found {} italian resources", num_italian);
-            if best_num < num_italian {
-                best_num = num_italian;
-                best_version = Some(VanillaVersion::ITALIAN);
-            }
-        }
-        if num_polish > 0 {
-            info!("Found {} polish resources", num_polish);
-            if best_num < num_polish {
-                best_num = num_polish;
-                best_version = Some(VanillaVersion::POLISH);
-            }
+
+        let version = match (num_dutch, num_german, num_italian, num_polish, num_russian) {
+            (n, 0, 0, 0, 0) if n > 0 => Some(VanillaVersion::DUTCH),
+            (0, n, 0, 0, 0) if n > 0 => Some(VanillaVersion::GERMAN),
+            (0, 0, n, 0, 0) if n > 0 => Some(VanillaVersion::ITALIAN),
+            (0, 0, 0, n, 0) if n > 0 => Some(VanillaVersion::POLISH),
+            (0, 0, 0, 0, n) if n > 0 => Some(VanillaVersion::RUSSIAN),
```

Comment:
RUSSIAN_GOLD is not relevant for this? I noticed it's not in the old code either, but better ask.

Reply:
Tbh, I just reused the code that was there before. I'm not even sure what the `RUSSIAN_GOLD` version exactly refers to. It seems to work with the russian version from steam (I didn't do exhaustive testing wether it crashes somewhere in the game though)

Reply:
That version guessing code is based on the paths returned by this function:
https://github.com/ja2-stracciatella/ja2-stracciatella/blob/c3e9284a195ef38d774ecb4b3c0cf601576cb996/src/game/GameRes.cc#L156